config/Dockerfiles/JenkinsfileRelease

```diff
@@ -0,0 +1,182 @@
+
+        containers: [
+            // This adds the custom slave container to the pod.
+            // Must be first with name 'jnlp'
+            containerTemplate(name: 'jnlp',
+                image: DOCKER_REPO_URL + '/' + OPENSHIFT_NAMESPACE + '/jenkins-continuous-infra-slave:stable',
+                ttyEnabled: false,
+                alwaysPullImage: true,
+                args: '${computer.jnlpmac} ${computer.name}',
+                command: '',
+                workingDir: '/workDir')
+        ]
+)
+
+{
+    node(podName) {
+        ansiColor('xterm') {
+            timestamps {
+                // We need to set env.HOME because the openshift slave image
+                // forces this to /home/jenkins and then ~ expands to that
+                // even though id == "root"
+                // See https://github.com/openshift/jenkins/blob/master/slave-base/Dockerfile#L5
+                //
+                // Even the kubernetes plugin will create a pod with containers
+                // whose $HOME env var will be its workingDir
+                // See https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesLauncher.java#L311
+                //
+                env.HOME = "/root"
+                //
+                try {
+                    stage('checkout-linchpin') {
+                        deleteDir()
+                        pipelineUtils.setDefaultEnvVars()
+                        pipelineUtils.verifyPod(OPENSHIFT_NAMESPACE, env.NODE_NAME)
+                        dir(repoName) {
+
+                            withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: 'paas-bot', usernameVariable: 'GH_USER', passwordVariable: 'GH_TOKEN']]) {
+                                checkout([$class: 'GitSCM',
+                                    branches: [[name: env.ghprbActualCommit]],
+                                    doGenerateSubmoduleConfigurations: false,
+                                    extensions                       : [],
+                                    submoduleCfg                     : [],
+                                    userRemoteConfigs                : [
+                                        [refspec:
+                                            '+refs/heads/*:refs/remotes/origin/*  +refs/tags/*:refs/tags/*',
+                                            url: "https://" + GH_USER + ":" + GH_TOKEN + "@github.com/${env.ghprbGhRepository}"]
+                                        ]
+                                ])
+                            }
+
+                        }
+                    }
+                    stage('release-version') {
+                        dir(repoName) {
+
+                            string version = sh (
+                                script: "cat linchpin/version.py | grep __version__ | cut -d' ' -f3 | cut -d\"'\" -f2",
+                                returnStdout: true
```

Comment:
while this works it feels fragile to me and would rather cat out a python file to import this version file and simply print it.

cat << EOF >return_version
#!/usr/bin/python
import linchpin.version
print linchpin.version.__version__
EOF

Then the command would be:
string version = sh (
   script: "./return_version", returnStdout: true
)

Reply:
Actually, there's a setup.py way to get this, I'll adjust that.